layout/AddressesUsingIdsQuery.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const _ = require('lodash');
+const Query = require('./Query');
+
+const baseQuery = {
+  query: {
+    function_score: {
+      query: {
+        bool: {
+          minimum_number_should_match: 1,
+          should: [],
+          filter: {
+            bool: {
+              minimum_number_should_match: 1
+            }
+          }
+        }
+      },
+      functions: []
+    }
+  }
+};
+
+function createShould(layer, ids) {
+  // create an object initialize with terms.'parent.locality_id' (or whatever)
+  return _.set({}, ['terms', `parent.${layer}_id`], ids);
+}
+
+function createAddressShould(vs) {
+  const should = {
+    bool: {
+      _name: 'fallback.address',
```

Comment:
not a big deal, but calling these `fallback.*` is a bit misleading since it makes me think there's a primary condition somewhere else in the query that might result in this fallback result being ignored